### PR TITLE
[Batcher] Add Readiness and Liveness Health Probe

### PR DESCRIPTION
**Pull request by siddimore**
## Why are these changes needed?
Couple of times on Testnet, it was observed Batcher stalled, with batcher implemented as a long-running process, once batcher is stalled it would need manual intervention to restart the container in K8s

In order to automate container restarts in K8s an option is to implement
1. Readiness Probe (check if Readiness file exists)
2. Health Based file probe (Periodic file creation)

In K8s batcher will have corresponding  probes configured. Anytime a stall is identified in this long-running process. Kubernetes Liveness Probe will fail resulting in restarting of the Pod.

These are not exact values but a representation

         livenessProbe:
          exec:
            command:
            - sh
            - -c
            - test $(expr $(date +%s) - $(stat -c %Y /path/to/heartbeat/file)) -lt 45
            initialDelaySeconds: 120
            periodSeconds: 120

         readinessProbe:
          exec:
            command:
            - cat
            - /tmp/ready
          initialDelaySeconds: 120
          periodSeconds: 10

Note: This feature can be disabled easily by updating HelmChart. Hence did not add config parameter to batcher

## Kubernetes Batcher STS Example
1. Readiness Probe also validated first time application start
2. LivenessProbe checked every 120 seconds 

Screenshot 2024-01-24 at 6 44 41 PM

## Batcher with no stalls:
Screenshot 2024-01-29 at 2 00 15 PM

Events
Screenshot 2024-01-29 at 2 02 09 PM

HealthCheck
Screenshot 2024-01-29 at 2 01 49 PM

MT3 Endpt: https://batcher-mt3-health.eigenda-dev.eigenops.xyz/healthz

## Batcher with stalls:
Screenshot 2024-01-29 at 2 08 13 PM

Events
Screenshot 2024-01-29 at 2 08 29 PM

HealthCheck
Screenshot 2024-01-29 at 2 08 37 PM

## Currently Monitoring on K8s
1. Ensure no unnecessary restarts are induced because of the introduced logic for stalls
2. Pod is restarted only if Main is stalled

## HelmChart Changes
https://github.com/Layr-Labs/eigenda-devops/pull/264

## Checks

- [x] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/cmd/batcher/main.go [hunk 1]
```diff
@@ -43,15 +51,43 @@ func main() {
...
+		}
+	}()
+
+	if _, err := os.Create(healthProbePath); err != nil {
+		log.Printf("Failed to create readiness file: %v", err)
+	}
+	// Start File based liveness probe
+	go heartBeatMonitor(healthProbePath, probeChan, maxStallDuration)
+
+	ticker := time.NewTicker(118 * time.Second)
```
Comment (ian-shim) [inline disperser/cmd/batcher/main.go:67]: How did we come up with 118s?

File: disperser/cmd/batcher/main.go [hunk 2]
```diff
@@ -43,15 +51,43 @@ func main() {
...
-	if err != nil {
-		log.Fatalf("application failed: %v", err)
+	// Run the CLI app in a separate goroutine
+	go func() {
+		if err := app.Run(os.Args); err != nil {
+			errorChan <- err
+		}
+	}()
+
+	if _, err := os.Create(healthProbePath); err != nil {
```
Comment (ian-shim) [inline disperser/cmd/batcher/main.go:61]: should this be `readinessProbePath`?

File: disperser/cmd/batcher/main.go [hunk 3]
```diff
@@ -43,15 +51,43 @@ func main() {
...
+		}
 	}
-
-	select {}
 }
 
 func RunBatcher(ctx *cli.Context) error {
+
+	// Clean up readiness file
+	if err := os.Remove(readinessProbePath); err != nil && !os.IsNotExist(err) {
```
Comment (ian-shim) [inline disperser/cmd/batcher/main.go:87]: why do we only report failure when `os.IsNotExist(err)`? shouldn't we report whenver `err != nil?`

**Review by dmanc — APPROVED** [review]
Overall lgtm just a few comments

**Pull request merged**